src/item_contents_ui.cpp

```diff
@@ -0,0 +1,325 @@
+#include "item_contents_ui.h"
+
+#include "item.h"
+#include "output.h"
+#include "string_input_popup.h"
+
+item_contents_ui::item_contents_ui( const item *containing_item )
+    : pContaining_item( containing_item )
+{
+    std::list<const item *> tempList = pContaining_item->all_items_top();
+    for( const item *content : tempList ) {
+        vItemList.push_back( content );
+    }
+
+    vFilteredItemList = vItemList;
+
+    ctxt = input_context( "INVENTORY", keyboard_mode::keychar );
+
+    ctxt.register_action( "DOWN", to_translation( "Next item" ) );
+    ctxt.register_action( "UP", to_translation( "Previous item" ) );
+    ctxt.register_action( "PAGE_DOWN", to_translation( "Page down" ) );
+    ctxt.register_action( "PAGE_UP", to_translation( "Page up" ) );
+    ctxt.register_action( "QUIT", to_translation( "Cancel" ) );
+    ctxt.register_action( "TOGGLE_FAVORITE", to_translation( "Toggle favorite" ) );
+    ctxt.register_action( "HELP_KEYBINDINGS" );
+    ctxt.register_action( "ANY_INPUT" ); // For invlets
+    ctxt.register_action( "INVENTORY_FILTER" );
+    ctxt.register_action( "EXAMINE_CONTENTS" );
+    ctxt.register_action( "SCROLL_ITEM_INFO_UP" );
+    ctxt.register_action( "SCROLL_ITEM_INFO_DOWN" );
+}
+
+void item_contents_ui::draw_borders()
+{
+    //Header border
```

Comment:
Consider using `border_helper` from `output.h`.

Reply:
Done.  If border_helper is the new standard approach, then an update to crafting_gui should go on the to-do list, since that's what I based draw_borders off of.